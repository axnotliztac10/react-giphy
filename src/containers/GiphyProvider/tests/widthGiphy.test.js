import React from 'react';
import toJSON from 'enzyme-to-json';
import { shallow } from 'enzyme';

import {
  withGiphy,
  mapStateToProps,
  mapDispatchToProps
} from '../withGiphy';

describe('<withGiphy />', () => {
  const Wrapped = withGiphy(() => <div />);
  const props = {
    trendingList: [],
    searchList: [],
    searchPagination: {},
    trendingError: {},
    searchError: {},
    favorites: [],
    getTrending: () => {},
    search: () => {},
    searchFavorites: () => {},
    addRemoveFavorite: () => {}
  };
  const wrapper = shallow(<Wrapped {...props} />);

  it('should render correctly', () => {
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});

describe('should test mapDispatchToProps and mapStateToProps', () => {
  it('getTrending', () => {
    const dispatch = jest.fn();

    mapDispatchToProps(dispatch).getTrending();
    expect(dispatch.mock.calls[0][0]).toEqual({ type: 'GET_TRENDING_INIT'});
  });

  it('search', () => {
    const dispatch = jest.fn();

    mapDispatchToProps(dispatch).search();
    expect(dispatch.mock.calls[0][0]).toEqual({ type: 'SEARCH_INIT'});
  });

  it('searchFavorites', () => {
    const dispatch = jest.fn();

    mapDispatchToProps(dispatch).searchFavorites();
    expect(dispatch.mock.calls[0][0]).toEqual({ type: 'SEARCH_FAVORITES'});
  });

  it('addRemoveFavorite', () => {
    const dispatch = jest.fn();

    mapDispatchToProps(dispatch).addRemoveFavorite();
    expect(dispatch.mock.calls[0][0]).toEqual({ type: 'ADD_REMOVE_FAVORITE'});
  });

  it('mapStateToProps', () => {
    const initialState = {
      gyphyProvider: {
        favorites: [],
        trendingList: [],
        searchList: [],
        searchPagination: {
          currentSearch: '',
          itemsPerPage: 6,
          offset: 0
        },
        trendingError: null,
        searchError: null,
      }
    };

    expect(mapStateToProps(initialState).searchList).toEqual(
      initialState.gyphyProvider.searchList
    );
  });
});
