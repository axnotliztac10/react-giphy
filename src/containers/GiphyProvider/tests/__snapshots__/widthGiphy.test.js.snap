// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<withGiphy /> should render correctly 1`] = `
<Component
  favorites={Array []}
  onAddRemoveFavorite={[Function]}
  onSearch={[Function]}
  onSearchFavorites={[Function]}
  searchError={Object {}}
  searchList={Array []}
  searchPagination={Object {}}
  trendingError={Object {}}
  trendingList={Array []}
/>
`;
